src/services/evaluation-administration-service.ts

```diff
@@ -344,6 +345,67 @@ export const publish = async (id: number) => {
   )
 }
 
+export const reopen = async (id: number) => {
+  const evaluationAdministration = await EvaluationAdministrationRepository.getById(id)
+
+  if (evaluationAdministration === null) {
+    throw new CustomError("Id not found", 400)
+  }
+
+  if (
+    evaluationAdministration.status !== EvaluationAdministrationStatus.Closed &&
+    evaluationAdministration.status !== EvaluationAdministrationStatus.Published
+  ) {
+    throw new CustomError("Only closed and published statuses are allowed.", 400)
+  }
+
+  const evaluations = await EvaluationRepository.getAllByFilters({
+    evaluation_administration_id: evaluationAdministration.id,
+  })
+
+  for (const evaluation of evaluations) {
+    const evaluationRatings = await EvaluationRatingRepository.aggregateSumByEvaluationId(
+      evaluation.id,
+      {
+        score: true,
+      }
+    )
+
+    const data = {
+      zscore: 0,
+      weighted_score: 0,
```

Comment:
not sure but should this be weighted_zscore too?

Reply:
@emmaacha done ^^